conformance_tests/tools/metrics/src/test_metric.cpp

```diff
@@ -2063,29 +2077,51 @@ TEST_F(
-            groupInfo.metricGroupHandle, rawData, metricValues,
-            metricValueSets);
-
-        std::vector<zet_metric_handle_t> metricHandles;
-        lzt::metric_get_metric_handles_from_metric_group(
-            groupInfo.metricGroupHandle, metricHandles);
-        std::vector<zet_metric_properties_t> metricProperties(
-            metricHandles.size());
-        lzt::metric_get_metric_properties_for_metric_group(metricHandles,
-                                                           metricProperties);
-
-        lzt::metric_validate_streamer_marker_data(metricProperties,
-                                                  metricValues, metricValueSets,
-                                                  streamerMarkerValues);
+        const char *value_string =
+            std::getenv("LZT_METRIC_READ_DATA_MAX_DURATION_MS");
+        uint32_t max_duration_in_milliseconds = 10;
+        if (value_string != nullptr) {
+          uint32_t value = atoi(value_string);
+          max_duration_in_milliseconds =
+              value != 0 ? value : max_duration_in_milliseconds;
+          max_duration_in_milliseconds =
+              std::min(max_duration_in_milliseconds, 100u);
+        }
+        auto start_time = std::chrono::high_resolution_clock::now();
+        auto duration = std::chrono::milliseconds(max_duration_in_milliseconds);
+        uint32_t streamer_marker_values_index = 0;
+        while ((std::chrono::high_resolution_clock::now() - start_time) <
+                   duration &&
+               (streamer_marker_values_index < streamerMarkerValues.size())) {
+          std::vector<uint8_t> rawData{};
+          uint32_t rawDataSize{};
+          std::this_thread::sleep_for(std::chrono::milliseconds(1));
+          lzt::metric_streamer_read_data(metricStreamerHandle, rawDataSize,
+                                         &rawData);
+
+          if (rawDataSize == 0) {
+            continue;
+          }
+          rawData.resize(rawDataSize);
+          std::vector<zet_typed_value_t> metricValues;
+          std::vector<uint32_t> metricValueSets;
+          lzt::metric_calculate_metric_values_from_raw_data(
+              groupInfo.metricGroupHandle, rawData, metricValues,
+              metricValueSets);
+
+          std::vector<zet_metric_handle_t> metricHandles;
+          lzt::metric_get_metric_handles_from_metric_group(
+              groupInfo.metricGroupHandle, metricHandles);
+          std::vector<zet_metric_properties_t> metricProperties(
+              metricHandles.size());
+          lzt::metric_get_metric_properties_for_metric_group(metricHandles,
+                                                             metricProperties);
+
+          lzt::metric_validate_streamer_marker_data(
+              metricProperties, metricValues, metricValueSets,
+              streamerMarkerValues, streamer_marker_values_index);
+        }
+        EXPECT_EQ(streamerMarkerValues.size(), streamer_marker_values_index);
```

Comment:
Please add a comment about this EXPECT.
"Checking if .."

Reply:
Resolved. Thank you.